lambda_src/drivers/destination_s3.py

```diff
@@ -107,9 +109,11 @@ def write(
     )
 
     if not prefix:
-        prefixed_filename = f'{DATA_FOLDER_NAME}/{batch_id}/row-{row_index}.data.json'
+        prefixed_filename = f'{batch_id}-row-{row_index}.data.json'
```

Comment:
this logic repeats regardless of which of the three options is true. instead, you can set this variable and then re-use it in the below.